src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -0,0 +1,303 @@
+}
+
+void GcsActor::UpdateAddress(const rpc::Address &address) {
+  actor_table_data_.mutable_address()->CopyFrom(address);
+}
+
+const rpc::Address &GcsActor::GetAddress() const { return actor_table_data_.address(); }
+
+WorkerID GcsActor::GetWorkerID() const {
+  const auto &address = actor_table_data_.address();
+  if (address.worker_id().empty()) {
+    return WorkerID::Nil();
+  }
+  return WorkerID::FromBinary(address.worker_id());
+}
+
+void GcsActor::UpdateState(rpc::ActorTableData::ActorState state) {
+  actor_table_data_.set_state(state);
+}
+
+rpc::ActorTableData::ActorState GcsActor::GetState() const {
+  return actor_table_data_.state();
+}
+
+ActorID GcsActor::GetActorID() const {
+  return ActorID::FromBinary(actor_table_data_.actor_id());
+}
+
+TaskSpecification GcsActor::GetCreationTaskSpecification() const {
+  const auto &task_spec = actor_table_data_.task_spec();
+  return TaskSpecification(task_spec);
+}
+
+const rpc::ActorTableData &GcsActor::GetActorTableData() const {
+  return actor_table_data_;
+}
+
+rpc::ActorTableData *GcsActor::GetMutableActorTableData() { return &actor_table_data_; }
+
+void GcsActor::AsyncFlush(std::function<void()> &&callback) {
+  RAY_LOG(INFO) << "Updating the state of actor " << GetActorID() << " to "
+                << ActorTableData::ActorState_Name(GetState()) << " with node id "
+                << GetNodeID();
+  auto actor_table_data = std::make_shared<rpc::ActorTableData>();
+  actor_table_data->CopyFrom(GetActorTableData());
+
+  auto actor_id = GetActorID();
+  auto node_id = GetNodeID();
+  auto state = GetState();
+  auto on_done = [actor_id, node_id, state, callback](Status status) {
+    if (!status.ok()) {
+      // TODO(ZhuSenlin): flush with retry?
+      RAY_LOG(FATAL) << "Failed to update the state of actor " << actor_id << " to "
+                     << ActorTableData::ActorState_Name(state) << " with node id "
+                     << node_id;
+    }
+    RAY_LOG(INFO) << "Succeeded in updating the state of actor " << actor_id << " to "
+                  << ActorTableData::ActorState_Name(state) << " with node id "
+                  << node_id;
+    if (callback) {
+      callback();
+    }
+  };
+  auto status =
+      redis_gcs_client_->Actors().AsyncUpdate(GetActorID(), actor_table_data, on_done);
+  if (!status.ok()) {
+    on_done(status);
+  }
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+GcsActorManager::GcsActorManager(
+    boost::asio::io_context &io_context, std::shared_ptr<gcs::RedisGcsClient> gcs_client,
+    gcs::GcsNodeManager &gcs_node_manager,
+    std::unique_ptr<gcs::GcsActorScheduler> &&gcs_actor_scheduler)
+    : redis_gcs_client_(std::move(gcs_client)),
+      gcs_actor_scheduler_(std::move(gcs_actor_scheduler)) {
+  RAY_LOG(INFO)
+      << "Initializing GcsActorManager and load actor information from storage.";
+  auto actor_table_data_list = LoadAllActorTableData();
```

Comment:
> Please remove this.
> 
> For the first implementation, we should make the GCS service non-persistent. This is not a regression since we currently do not support this functionality with the pure redis GCS. It also reduces the amount/complexity of logic that we have to review at once.

Okey, I will remove it.